core/arch/arm/tee/tee_rpmb.c

```diff
@@ -804,18 +863,36 @@ static TEE_Result tee_rpmb_init_read_wr_cnt(uint16_t dev_id,
 	return res;
 }
 
+static TEE_Result tee_rpmb_verify_key_sync_counter(uint16_t dev_id)
+{
+	uint16_t op_result = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	res = tee_rpmb_init_read_wr_cnt(dev_id, &rpmb_ctx->wr_cnt,
+					&op_result);
+
+	if (res == TEE_SUCCESS) {
+		rpmb_ctx->key_verified = true;
+		rpmb_ctx->wr_cnt_synced = true;
+	}
+
+	IMSG("Verify key returning 0x%x\n", res);
+	return res;
+}
+
 /*
  * This function must never return TEE_SUCCESS if rpmb_ctx == NULL.
  */
 static TEE_Result tee_rpmb_init(uint16_t dev_id, bool writekey, bool commercial)
 {
-	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_Result res = TEE_SUCCESS;
```

Comment:
We usually check the presence of relevant crypto functions before using them, somewhere in this function would be convenient to do those checks.